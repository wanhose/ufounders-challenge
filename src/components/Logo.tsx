import { SVGAttributes } from 'react';
import { ReactComponent as Isologotype } from 'assets/brand/isologotype.svg';
import { ReactComponent as Isotype } from 'assets/brand/isotype.svg';

type Props = SVGAttributes<SVGSVGElement> & {
  variant?: 'isologotype' | 'isotype';
};

const Logo = ({ variant, ...props }: Props) => {
  switch (variant) {
    case 'isologotype':
      return <Isologotype {...props} />;
    case 'isotype':
    default:
      return <Isotype {...props} />;
  }
};

export default Logo;

export type LogoProps = Props;
