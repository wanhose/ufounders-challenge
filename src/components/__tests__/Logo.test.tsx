import { render } from 'tests/utils';

import Logo from '../Logo';

describe('Logo', () => {
  it('isologotype variant renders correctly', () => {
    const { container } = render(<Logo variant="isologotype" />);

    expect(container).toMatchSnapshot('Isologotype');
  });

  it('isotype variant renders correctly', () => {
    const { container } = render(<Logo variant="isotype" />);

    expect(container).toMatchSnapshot('Isotype');
  });
});
