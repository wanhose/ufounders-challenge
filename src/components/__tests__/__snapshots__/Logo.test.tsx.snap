// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo isologotype variant renders correctly: Isologotype 1`] = `
<div>
  <svg>
    isologotype.svg
  </svg>
</div>
`;

exports[`Logo isotype variant renders correctly: Isotype 1`] = `
<div>
  <svg>
    isotype.svg
  </svg>
</div>
`;
